test/es-module/test-cjs-legacyMainResolve.js

```diff
@@ -150,7 +150,7 @@ describe('legacyMainResolve', () => {
     );
     assert.throws(
       () => legacyMainResolve(packageJsonUrl, { main: './index.node' }, packageJsonUrl),
-      { message: /index\.node/, code: 'ERR_MODULE_NOT_FOUND' },
+      { message: /No package entry point defined for package/, code: 'ERR_MODULE_NOT_FOUND' },
```

Comment:
IIUC, there is an entry point defined, `'./index.node'`, but the resolution still fails because the file doesn't exist. This change would likely be confusing.

Reply:
yeah make sense, so I'm gonna check wheher user request a file or a module then give a differnt error